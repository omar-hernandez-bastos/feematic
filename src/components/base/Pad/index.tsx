import React, { ButtonHTMLAttributes } from 'react'

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
  text: string | React.ReactNode
  className?: string
  isActive: boolean
}

const Pad = ({ text, className, isActive, ...props }: Props) => {
  return (
    <div className={className}>
      <button
        {...props}
        className={`rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center ${
          isActive ? 'shadow-2xl shadow-cyan-500/50' : ''
        }`}
      >
        {text}
      </button>
    </div>
  )
}
Pad.defaultProps = {
  className: ''
}
export default Pad
