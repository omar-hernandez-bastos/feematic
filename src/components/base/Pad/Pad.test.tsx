import { render } from '@testing-library/react'
import Pad from '.'

describe('Pad Test', () => {
  const padProps = {
    text: '1',
    className: '',
    isActive: false
  }
  const { container } = render(<Pad {...padProps} />)
  expect(container).toMatchSnapshot()
})
