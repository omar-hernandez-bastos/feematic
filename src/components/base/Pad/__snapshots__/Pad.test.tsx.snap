// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pad Test Pad Component should render correctly 1`] = `
<div>
  <div
    class=""
  >
    <button
      class="rounded-2xl  text-4xl bg-gray-900 text-cyan-500 w-full h-full grid place-items-center "
    >
      1
    </button>
  </div>
</div>
`;
